app/scripts/modules/core/pipeline/config/stages/deploy/deployExecutionDetails.controller.js

```diff
@@ -58,10 +63,56 @@ module.exports = angular.module('spinnaker.core.pipeline.stage.deploy.details.co
         }
       }
       $scope.deployed = results;
+      configureWaitingMessages(results);
       $scope.provider = context.cloudProvider || context.providerType || 'aws';
     };
 
-    this.overrideFiltersForUrl = clusterFilterService.overrideFiltersForUrl;
+    function configureWaitingMessages(deployedArtifacts) {
+      $scope.showWaitingMessage = false;
+      $scope.waitingForUpInstances = false;
+      $scope.showScalingActivitiesLink = false;
+      $scope.showPlatformHealthOverrideMessage = false;
+
+      if (!deployedArtifacts.length) {
+        return;
+      }
+      const deployed = deployedArtifacts[0];
+      const stage = $scope.stage;
+
+      const activeWaitTask = (stage.tasks || []).find(t => ['RUNNING', 'TERMINAL'].includes(t.status) && t.name === 'waitForUpInstances');
+
+      if (activeWaitTask && stage.context.lastCapacityCheck) {
+        $scope.showWaitingMessage = true;
+        $scope.waitingForUpInstances = activeWaitTask.status === 'RUNNING';
+        const lastCapacity = stage.context.lastCapacityCheck;
+        const waitDurationExceeded = activeWaitTask.runningTimeInMs > (3 * 60 * 1000);
```

Comment:
what do you think about doing something like `moment.duration(3, 'minutes').asMilliseconds()` to make stuff like this more readable?  I tried it in the cache factory rewrite and like it.  if we're going to use moment, let's use moment.  thoughts?